### ISSUE TITLE
FP S2589 with pattern matching to bool

### ISSUE THREAD

**Issue by KarbyCZ** [SonarSource/sonar-dotnet#3222]
### Description

S2589 gives a false positive for pattern matching a boolean and then evaluation of that value

### Repro steps

```csharp
object x = false;

if ((x is bool b) && b)
    Console.WriteLine("It is true");
else
    Console.WriteLine("It is something else");
```

### Expected behavior

The rule S2589 should not fire in this case, because the value could be either true or false.

### Actual behavior

The rule is pointing to the right side of && where the boolean b value is evaluated. The message is

> Change this condition so that it does not always evaluate to 'true'.

### Related information

* SonarAnalyzer.CSharp version 8.5.0.15942
* VS2019 version 16.4.6
* MSBuild version 16.4.0, dotnet SDK version 3.1.102
* Windows 10, build 1909

**Comment by KarbyCZ** [comment]
Duplicate of #3123
